### PR TITLE
feat: Adds the Featured Charts dashboard

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
Creates a new example dashboard called Featured Charts with the objective of demonstrating our most active supported charts. Another objective is to provide a dashboard where we can test our plugin features using these chart types.

Screenshot 2024-05-31 at 09 27 47
Screenshot 2024-05-31 at 09 28 10
Screenshot 2024-05-31 at 09 28 32
Screenshot 2024-05-31 at 09 28 57

<br>This PR also changes the viz gallery to include missing featured charts and sort the charts alphabetically to help users find the charts they are looking for. Another small tweak was to add some margin to the Create New Chart button.

Screenshot 2024-05-31 at 10 36 49

### TESTING INSTRUCTIONS
Check the Featured Charts dashboard and the viz gallery changes.

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by michael-s-molina** [comment]
@rusackas @yousoph @kasiazjc @kgabryje Please review the viz gallery changes. Specifically, if you agree to sort the charts alphabetically.

FYI: We need https://github.com/apache/superset/pull/28652 for the Histogram chart. Right now it will render with an error.

**Comment by john-bodley** [comment]
@michael-s-molina I love the color palette. Have you considered updating the viz gallery thumbnails with your examples so it feels more coherent/consistent—upleveling the visual quality?

**Comment by michael-s-molina** [comment]
> @michael-s-molina I love the color palette. Have you considered updating the viz gallery thumbnails with your examples so it feels more coherent/consistent—upleveling the visual quality?

That's a great suggestion! What do you think @kasiazjc @rusackas?

File: superset/examples/configs/charts/Featured Charts/Graph_124.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  target: product_line
+  metric: count
+  adhoc_filters:
+    - clause: WHERE
+      subject: order_date
+      operator: TEMPORAL_RANGE
+      comparator: No filter
+      expressionType: SIMPLE
+  row_limit: 10000
+  color_scheme: supersetColors
```
Comment (john-bodley) [inline superset/examples/configs/charts/Featured Charts/Graph_124.yaml:35]: For some reason the color palette seems different for this chart in your screenshot.

Comment (michael-s-molina) [reply superset/examples/configs/charts/Featured Charts/Graph_124.yaml:35]: This is the chart's default color palette. My screenshot is using the dashboard's color palette which overrides every chart in it.

File: superset/examples/configs/dashboards/Featured_Charts_15.yaml
Comment (rusackas) [inline superset/examples/configs/dashboards/Featured_Charts_15.yaml:1]: Maybe worth stripping the `_15` from the filename. It doesn't hurt, but there's no real reason to have it either. You can change the filename without breaking anything here.

**Comment by michael-s-molina** [comment]
> @michael-s-molina I love the color palette. Have you considered updating the viz gallery thumbnails with your examples so it feels more coherent/consistent—upleveling the visual quality?

@john-bodley I opened https://github.com/apache/superset/discussions/29076 to register your suggestion and added a topic for the next Town Hall.

File: superset/examples/configs/dashboards/Featured_Charts_15.yaml
Comment (michael-s-molina) [reply superset/examples/configs/dashboards/Featured_Charts_15.yaml:1]: I removed the ID from all assets.

**Review by john-bodley — APPROVED** [review]

**Review by rusackas — APPROVED** [review]
LGTM!

**Pull request merged**